modules/ROOT/pages/authentication-authorization/sso-integration.adoc

```diff
@@ -253,6 +253,109 @@ dbms.security.oidc.mysso.claims.groups=roles
 
+
+[role=label--new-5.24]
+[[auth-sso-auth-objects]]
+=== Use Auth Objects to configure SSO at the user level
+xref:authentication-authorization/auth-objects.adoc[Auth Objects] can be used to determine which users can authenticate and authorize using the configured providers.
+
+[NOTE]
+====
+You need to set the `dbms.security.require_local_user` configuration setting to `true` to use Auth Objects. This means that a user with a matching Auth Object *must* exist in order to be able to authenticate and authorize.
+====
+
+The following examples show how you can use Cypher to do this.
+
+
+.Create a user with an Auth Object who can authenticate and authorize using `oidc-mysso`.
+======
+----
+CREATE USER jake
+SET AUTH 'oidc-mysso' {SET ID 'jakesUniqueMySso1Id'} // this must match the claim that you configured via dbms.security.oidc.mysso.claims.username
+----
+This will create a user who can authenticate and authorize using `mysso` if they present a valid token with a `sub` claim of `jakesUniqueMySso1Id`. The claim used for authentication is determined by the xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.claims.username[`dbms.security.oidc.mysso.claims.username`] config setting (default the default is the `sub` claim).
+
+When the `dbms.security.require_local_user` configuration setting is set to `true`, users can *only* authenticate when there is a user in the database with an Auth Object which links to the provider that the user is trying to authenticate with. If there is no matching Auth Object, the user will not be able to authenticate or authorize. This applies to all providers.
+======
+
+
+.Create a user with two Auth Objects allowing the user to authenticate and authorize with either of two providers.
+======
+----
+CREATE USER jake
+SET HOME DATABASE anotherDb
+SET AUTH 'oidc-mysso1' {SET ID 'jakesUniqueMySso1Id'} // this must match the claim that you configured via dbms.security.oidc.mysso1.claims.username
+SET AUTH 'oidc-mysso2' {SET ID 'jakesUniqueMySso2Id'} // this must match the claim that you configured via dbms.security.oidc.mysso2.claims.username
+----
+This will create a user who can authenticate and authorize using `oidc-mysso1` or `oidc-mysso2`.
+This example also illustrates that the user can have their `HOME` database set even when using an external auth provider.
+======
+
+
+.Alter a user to remove one of their Auth Objects.
+======
+----
+ALTER USER jake
+REMOVE AUTH 'oidc-mysso2'
+
+----
+This will prevent the user from being able to authenticate and authorize with the `oidc-mysso2` provider.
+======
+
+.Alter a user to allow them to authenticate and authorize using username and password
+======
+----
+ALTER USER jake
+SET AUTH 'native' {SET PASSWORD 'changeme' SET PASSWORD CHANGE REQUIRED}
+
+----
+This will allow the user to authenticate and authorize using the specified username and password
```

Comment:
```suggestion
This will allow the user to authenticate and authorize using the specified username and password (in addition to what they are already configured to use).
```